src/hooks.rs

```diff
@@ -297,6 +605,84 @@ impl InnerConnection {
             }
         };
     }
+
+    pub fn authorizer<'c, F>(&'c mut self, authorizer: Option<F>)
+    where
+        F: for<'r> FnMut(AuthContext<'r>) -> Authorization + Send + RefUnwindSafe + 'static,
+    {
+        unsafe extern "C" fn call_boxed_closure<'c, F>(
+            p_arg: *mut c_void,
+            action_code: c_int,
+            action_arg1_str: *const c_char,
+            action_arg2_str: *const c_char,
+            db_str: *const c_char,
+            accessor_str: *const c_char,
+        ) -> c_int
+        where
+            F: FnMut(AuthContext<'c>) -> Authorization + Send + 'static,
+        {
+            use std::ffi::CStr;
+            use std::str;
+
+            let optional_str = |p_str: *const c_char| {
+                if p_str.is_null() {
+                    None
+                } else {
+                    let c_slice = CStr::from_ptr(p_str).to_bytes();
+                    str::from_utf8(c_slice).ok()
```

Comment:
If any string parameter is not utf-8, `None` value makes `AuthAction::from_raw` return `Unknown`.

Reply:
want it to `expect` instead?